### DISCUSSION TITLE
Input Group

### DISCUSSION THREAD

**Discussion by Judge-Paul** [shadcn-ui/ui#2566] [Ideas]
I am thinking you can create an Input group component so we can build things like
![image](https://github.com/shadcn-ui/ui/assets/110723341/6484d8b6-60e3-48ea-9b74-f0d18befa1fe)
By just adding the icon and input type

**Comment by NjengaFelix** [comment]
The shadcn-ui [landing page](https://ui.shadcn.com) has an example of this search input.  

I have highlighted the search input below
![image](https://github.com/shadcn-ui/ui/assets/32095327/a4c82867-729a-4e48-8a16-ef5af04cdcbf)

I have shared the code snippet of the search input field from the [mail component example in shadcn-ui github examples](https://github.com/shadcn-ui/ui/blob/main/apps/www/app/examples/mail/components/mail.tsx)


```
import {
  Search,
} from "lucide-react"

<div className="relative">
     <Search className="absolute left-2 top-2.5 h-4 w-4 text-muted-foreground" />
      <Input placeholder="Search" className="pl-8" />
</div>
```


They have simply used a lucide icon and passed the relevant styles using tailwind. 

I am therefore assuming, this input can't be supported because:
1. The search input icon varies
2. They wouldn't want to support icons that require importation.

My response is based on assumptions but not facts

**Comment by Judge-Paul** [comment]
Thanks for this, this is exactly what I was looking for

**Comment by arnilrashid** [comment]
https://github.com/arnilrashid/SearchGroup-react-js
